django/tasks/base.py

```diff
@@ -0,0 +1,83 @@
+from __future__ import unicode_literals
+
+from django.conf import settings
+from django.core.exceptions import ImproperlyConfigured
+from django.utils.module_loading import import_string
+
+from .exceptions import InvalidTaskBackendError
+
+DEFAULT_TASK_ALIAS = 'default'
+
+# backends cache to make sure we don't create one per connection
+# all the backends should be thread-safe
+backends = {}
+
+
+def _create_task_backend(alias):
+    try:
+        conf = settings.QUEUES[alias]
+    except KeyError:
+        if alias != DEFAULT_TASK_ALIAS:
```

Comment:
Yes, but the implementation is very different (and better) for the template fragment cache. It's the _user_ (the template fragment cache template tag) that checks whether the specific `template_fragments` cache exists and falls back to default if not. The core get-cache-backend API doesn't implement this silent fallback unconditionally. And if you explicitly specify `using=...` with the template fragment cache tag, it's an error if the cache you specify doesn't exist, there is no fallback.

Better API options to implement the use case you're aiming for:

1) If it's an unusual case, just have the admin itself check whether there is an `email` queue, and fallback if not.

2) If we want this to be a more generic feature, allow specifying a list of backends rather than just a single backend, so it will fallback through that list, but still error if nothing in the list is found.

3) Or (a simplification of (2)), add a keyword argument something like `fallback_to_default` that defaults to `False` but can be set to `True` if you want to fallback to default if the given queue is not found.

IMO any of those could be acceptable, but unconditional silent fallback to default, without any way to override it and say "No I really mean I want this queue, it should be an error if it doesn't exist," is not.